src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -491,13 +538,28 @@
     <Message Importance="normal" Text="Sonar: ($(MSBuildProjectFile)) Analysis configured successfully with $(SonarProjectConfigFilePath)." />
   </Target>
 
-  <Target Name="SetRazorCodeAnalysisProperties" BeforeTargets="RazorCoreCompile" AfterTargets="CopyFilesToOutputDirectory">
+  <!-- **************************************************************************** -->
+  <!-- This target is only run in case a Razor compilation is happening.
+       This target sets the Razor specific error log properties and copies the results of the core compilation.
+       The *.pb files will be copied to ProjectSpecificOutDir.tmp directory as during the RazorCoreCompilation the *.pb
+       files will be replaced by the ones for Razor.
+
+       We want to run after the core "RazorCoreCompile" to copy the content before it is overwriten.
```

Comment:
But this target runs `BeforeTargets="RazorCoreCompile"`. That is confusing. Let's just remove/simplify what's not needed and keep only the big picture here.